src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { Checkbox as CoreCheckbox } from 'wix-ui-core/checkbox';
+import styles from './Rating.st.css';
+import { ReactComponent as StarIcon } from '../../assets/icons/Star.svg';
+
+export enum IconSize {
+  LARGE = 'large',
+}
+
+export enum Mode {
+  EDIT = 'edit',
+  PREVIEW = 'preview',
+}
+
+export interface RatingProps {
+  value: number;
+  mode: Mode;
+  onSelect?(value: number): void;
+  maxRating?: number;
+  disabled?: boolean;
+  error?: boolean;
+  iconSize?: IconSize;
+}
+
+interface DefaultProps {
+  value: number;
+  maxRating: number;
+  disabled: boolean;
+  error: boolean;
+  mode: Mode;
+}
+
+/** Rating component based on IconToggle */
+export class Rating extends React.Component<RatingProps> {
+  static displayName = 'Rating';
+  static defaultProps: DefaultProps = {
+    value: 0,
+    maxRating: 5,
+    disabled: false,
+    error: false,
+    mode: Mode.PREVIEW,
+  };
+
+  _renderContent = () => <StarIcon />;
+
+  render() {
+    const {
+      value,
+      maxRating,
+      onSelect,
+      disabled,
+      error,
+      iconSize,
+      mode,
+      ...rest
+    } = this.props;
+    const content = this._renderContent();
+    const ratingList = Array.from(new Array(maxRating));
+    const ratingListLength = ratingList.length;
+
+    return (
+      <div {...styles('root', {}, rest)}>
+        <div className={styles.iconList}>
+          {ratingList.map((_el, idx: number) => {
+            const checked = ratingListLength - idx <= value;
+
+            return (
+              <span
+                key={idx}
+                {...styles('icon', {
+                  checked,
+                  disabled,
+                  error,
+                  iconSize,
+                  mode,
+                })}
+              >
+                <CoreCheckbox
```

Comment:
Very interesting approach for the component  🤔 
I haven't thought about this kind of implementation.

However, i'm not sure checkboxes are the way to go, in an a11y point of view.
The correct way would probably be using something like a radio group.
You can see what they suggest here: https://www.w3.org/WAI/tutorials/forms/custom-controls/#a-star-rating

I'm not sure we really have to use the radio buttons, but maybe we'll have to pass the correct `aria-*` attributes.

Reply:
OK, I'll change it to `radio group`. 
I was a bit confused cs in docs was mention that this component should be based on `IconToggle` and this component implemented on `CoreCheckbox` base.